src/main/java/reactor/core/publisher/FluxMergeSequential.java

```diff
@@ -128,18 +142,16 @@ public void subscribe(Subscriber<? super R> s) {
 		static final AtomicLongFieldUpdater<MergeSequentialMain> REQUESTED =
 				AtomicLongFieldUpdater.newUpdater(MergeSequentialMain.class, "requested");
 
-		public MergeSequentialMain(Subscriber<? super R> actual,
+		MergeSequentialMain(Subscriber<? super R> actual,
 				Function<? super T, ? extends Publisher<? extends R>> mapper,
-				int maxConcurrency, int prefetch, ErrorMode errorMode) {
-
+				int maxConcurrency, int prefetch, ErrorMode errorMode,
+				Supplier<Queue<MergeSequentialInner<R>>> queueSupplier) {
 			this.actual = actual;
 			this.mapper = mapper;
 			this.maxConcurrency = maxConcurrency;
 			this.prefetch = prefetch;
 			this.errorMode = errorMode;
-			this.subscribers = QueueSupplier.<MergeSequentialInner<R>>get(Math.min(
-					prefetch,
-					maxConcurrency)).get();
+			this.subscribers = queueSupplier.get();
```

Comment:
In RxJava 2, the subscribes is an unbouded queue with the island size of the smaller value because sometimes `maxConcurrency` could be `Integer.MAX_VALUE` yet `prefetch` is reasonable.